### DISCUSSION TITLE
Lenny board update to PIC32MX274

### DISCUSSION THREAD

**Discussion by cb88** [chipKIT32/chipKIT-core#466] [Ideas]
It appears the XLP chips may have fewer erratta... I seem to remember Matt saying there were some changes needed to adapt Lenny to the XLP chip. Currently two issues with PIC32MX270 affect me, serial overflows and reset via program being broken I'm not sure if updating to MX274 would fix this but it would be nice, in addition to lower power use.

**Comment by majenkotech** [comment]
It has another benefit: clockless USB operation. We can use the internal oscillator synced with the USB clock instead of needing an external oscillator to power it - makes for a simpler / smaller board.

I really do want to get the 274 running, but the bootloader has been a sticking point so far. It's not something I have looked at for a while though, but I do keep stumbling across my test board and thinking "I really must get to grips with that again".

I have a feeling it may not be until after I have managed to get the new compiler working though, and heaven only knows when that'll be.

**Reply by cb88** [reply]
I think in my case I would still need a clock as I have several boards that communicate via IR so they need stable clocks and can't sync to the USB clock (I think that is how that works for clockless USB IIRC). I've been using a MEMS SiTime oscillator which hasn't caused me any issues yet it just works until you put it in a room full of helium!

**Reply by majenkotech** [reply]
The internal oscillator is more than stable enough for simple IR communication (you're talking serial baud rate speeds here). USB needs much much tighter timing than simple serial / IRDA / similar. For that the internal PLL synchronises with the USB's timing to make the internal oscillator more precise than it already is. 

For FS USB 2.0 your USB clock needs to be 48MHz ± 0.25%. For UART communication you can drift by tens of percents and still be fine. For the '274 the internal oscillator is 0.9% accurate (pre-tuning by the USB clock). More than enough for simple serial communication.

**Comment by cb88** [comment]
MX274 support sure would be nice about now... *glances at low stock of the MX270* ... I wonder if automotive companies are gobbling it up or if they are making fewer of them.

**Reply by majenkotech** [reply]
In general it's lower production because they are focussing the fabs on making the chips the automotive companies want.

**Reply by cb88** [reply]
Yeah I know.... its a shame everybody's "heated seats" are sucking the fabs dry though.
